### DISCUSSION TITLE
Does Tasmota officially support WPA3 ?

### DISCUSSION THREAD

**Discussion by FaserF** [arendst/Tasmota#13414] [Q&A] [answered]
I am wondering as I have seen some ESP projects where WPA3 support was enabled.

**Comment by Jason2866** [comment, marked as answer]
No. We are going to test with ESP32. Standard Arduino ESP32 does not support. It needs a config change in ESP IDF and custom compiled libs for Arduino. Afaik there is no support for WPA3 with Esp8266
Edit: There will be no support of WPA3 and Tasmota with ESP8266. Reason the used ESP8266 SDK in Arduino does not support.  

EDIT (January 2022):
All Tasmota builds for ESP32, ESP32-C3 and ESP32-S2 do support WPA3

**Comment by Jason2866** [comment]
Builded today a test ESP32 Arduino framework with enabled WPA3 support. A Arduino ESP32 test sketch did work
```
rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DOUT, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12352
load:0x40080400,len:2912
entry 0x400805c4
Attempting to connect to SSID: TestWPA3
.................Connected to TestWPA3
``` 
Stay tuned...

**Reply by kroon040** [reply]
@Jason2866 , can you post also the test sketch so I can also test with this.

**Reply by Jason2866** [reply]
Did not save the small sketch. It was a few lines standard example i quick googled...
Something like this https://www.arduino.cc/en/Reference/WiFiClient
There is no setting needed. All standard. Connection time takes a bit longer.

**Reply by kroon040** [reply]
ok, there is no setting in the code for wpa3?  using this code, I should get the return the right encryption type...?

https://www.arduino.cc/en/Reference/WiFiEncryptionType

**Reply by Jason2866** [reply]
If espressif did it implemented in Arduino ESP32, yes. Never tried...
Let us know if it works. Thx.

**Reply by kroon040** [reply]
I go try, also nice feature to have in tasmota that you can see if you are connect to wpa2/wpa3 network

**Reply by kroon040** [reply]
found this, so there should be something in the framework

https://github.com/espressif/arduino-esp32/issues/2928

**Comment by Jason2866** [comment]
With PR #13526 there is the Tasmota32 build variant `tasmota32-dev` which uses the actual Tasmota32 Arduino development framework. This framework does support WPA3.
Tasmota connected to my WPA3 AP successfully  using this settings `SO56 0, SO57 0, wificonfig 5`

**Reply by kroon040** [reply]
The development are these ? (because I don't see an tasmota-dev) So I can test with these with wpa3?

http://ota.tasmota.com/tasmota32/

**Reply by sfromis** [reply]
There is no precompiled build for this variant, you select `tasmota32-dev` when building yourself.

**Reply by kroon040** [reply]
Hi I try to complile agait to test a new build, but there is no dev anymore?

platformio run -e tasmota32-dev
Error: Unknown environment names 'tasmota32-dev'. Valid names are 'tasmota-GR, tasmota-RU, tasmota32-core2, tasmota32-KO, tasmota32-lvgl, tasmota32-NL, tasmota32-CN, tasmota-CZ, tasmota-PT, tasmota-knx, tasmota-UK, tasmota32-display, tasmota-zigbee, tasmota32-FR, tasmota32_base, tasmota32-odroidgo, tasmota32-ir, tasmota-IT, tasmota32-BG, tasmota-ES, tasmota-VN, tasmota, tasmota-TR, tasmota32-PL, tasmota-lite, tasmota-ir, tasmota32-bluetooth, tasmota32-RO, tasmota32-TW, tasmota-display, tasmota-NL, tasmota-FR, tasmota-DE, tasmota32-webcam, tasmota32-TR, tasmota-BR, tasmota32_16M, tasmota-PL, tasmota-FY, tasmota32-HU, tasmota-minimal, tasmota32-SE, tasmota32-SK, tasmota32-PT, tasmota-CN, tasmota32-GR, tasmota32-BR, tasmota32-ES, tasmota32c3, tasmota-TW, tasmota-SK, tasmota32-HE, tasmota32_8M, tasmota32-CZ, tasmota32-DE, tasmota-KO, tasmota32-FY, tasmota-SE, tasmota-zbbridge, tasmota32, tasmota32-AF, tasmota32solo1, tasmota32-IT, tasmota-AF, tasmota-RO, tasmota32-UK, tasmota-BG, tasmota32-VN, tasmota-sensors, tasmota32-RU, tasmota-HU, tasmota-HE'

**Comment by Jason2866** [comment]
Now it is ☺ https://github.com/Jason2866/Tasmota-specials/raw/firmware/firmware/tasmota32/other/tasmota32-dev.bin
You can flash via Webbrowser https://jason2866.github.io/Tasmota-specials/

**Comment by kroon040** [comment]
Hi,

I used your tasmota32-dev.bin

Connected first to my wpa2 network that works. dit the setoption 56 and 57 to 0.

Enter wificonfig 5, and said wait?

00:01:51.901 CMD: wificonfig 5
00:01:51.914 MQT: stat/test/RESULT = {"WifiConfig":{"5":"Wait"}}

Tried connecting to my wpa3 network and that fails

00:00:09.164 WIF: Connect failed with AP timeout
00:00:12.467 WIF: Connecting to AP1 testje in mode 11n as esp32cam...
00:00:15.930 WIF: Connect failed with AP timeout
00:00:19.075 WIF: Connect failed with AP timeout
00:00:20.568 WIF: Connecting to AP1 testje in mode 11n as esp32cam...
00:00:23.939 WIF: Connect failed with AP timeout
00:00:27.106 WIF: Connect failed with AP timeout
00:00:28.588 WIF: Connecting to AP1 testje in mode 11n as esp32cam...
00:00:31.946 WIF: Connect failed with AP timeout
00:00:35.133 WIF: Connect failed with AP timeout
00:00:36.587 WIF: Connecting to AP1 testje in mode 11n as esp32cam...
00:00:39.931 WIF: Connect failed with AP timeout
00:00:43.096 WIF: Connect failed with AP timeout
00:00:44.587 WIF: Connecting to AP1 testje in mode 11n as esp32cam...
00:00:47.935 WIF: Connect failed with AP timeout
00:00:53.918 WIF: Connect failed with AP timeout
00:00:55.658 WIF: Connecting to AP1 testje in mode 11n as esp32cam...
00:00:58.907 WIF: Connect failed with AP timeout

Setting thje ssid back to wpa2 than it works again.

**Reply by Jason2866** [reply]
Yes, dev removed since the core used in dev is now standard for all Tasmota32 env

**Reply by sfromis** [reply]
With dev tasmota32.bin, I can confirm to be able to connect to an AP requiring WPA3, without any settings tweaks.

**Comment by kroon040** [comment]
Hi,

I had also an ap flashed with openwrt 21.02. When enabled wpa3 only there it works. I think the timing could be an issue with the other AP's because here you also see an reset and timout and than a connected. 

Is it possible to extend the time to connect?
```

[22:14:15]00:00:00.003 HDW: ESP32-D0WDQ6 
[22:14:15]00:00:00.036 UFS: FlashFS mounted with 304 kB free
[22:14:15]00:00:00.088 CFG: Loaded from File, Count 11
[22:14:15]00:00:00.104 QPC: Count 1
[22:14:15]00:00:00.205 BRY: Berry initialized, RAM used=4245
[22:14:15]00:00:00.222 BRY: no 'preinit.be'
[22:14:15]00:00:00.228 Project tasmota - Tasmota Version 10.0.0.3(tasmota)-2_0_1(2021-11-21T19:15:45)
[22:14:15]00:00:00.330 BRY: no 'autoexec.be'
[22:14:16]00:00:00.003 WIF: Connecting to AP1 wpa3test Channel 5 BSSId 34:FC:B9:37:E0:23 in mode 11n as tasmota-663600-5632...
[22:14:18]00:00:02.966 WIF: Connect failed as AP cannot be reached
[22:14:18]00:00:03.329 WIF: Connecting to AP1 wpa3test in mode 11n as tasmota-663600-5632...
[22:14:22]00:00:05.927 QPC: Reset
[22:14:24]00:00:08.946 WIF: Connect failed with AP timeout
[22:14:27]00:00:12.471 WIF: Connected
[22:14:28]22:14:27.218 HTP: Web server active on tasmota-663600-5632 with IP address 192.168.180.44
[22:14:29]22:14:28.760 RSL: INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"10.0.0.3(tasmota)","FallbackTopic":"cmnd/DVES_663600_fb/","GroupTopic":"cmnd/tasmotas/"}}
[22:14:29]22:14:28.775 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-663600-5632","IPAddress":"192.168.180.44"}}
[22:14:29]22:14:28.779 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU"}}
```

**Comment by schildbach** [comment]
Is there a chance WPA3 works on the Shelly Plus 1/1PM too, given that there is a different Tasmota flavor to use on these devies (tasmota32solo1.bin)?

**Reply by sfromis** [reply]
I had no trouble connecting a lamp with tasmota32solo1.bin to an AP requiring WPA3.

**Reply by Jason2866** [reply]
All ESP32xx variants have the same code base. For the solo1 the difference is just in sdkconfig to use one core when compiling the arduino libs. So yes WPA3 is supported.

**Comment by henkiejan1** [comment]
I could also say that WPA3 works with the ESP32 builds without any problem. Maybe change the selected anwser? it's there for the ESP32 and works! :)

**Comment by Jason2866** [comment]
If WPA3 is not working with the ESP32 and your Router / AP,  you have to change settings there.
**PMF (Protected Management Frames) is a prerequisite feature for a WPA3 connection, it needs to be explicitly configured before attempting connection.**

**Comment by BrAiNeeBug** [comment]
ESP8266 WPA3 and PMF Support !?
https://github.com/espressif/ESP8266_RTOS_SDK

**Reply by Jason2866** [reply]
No, Tasmota esp8266 is not based on RTOS SDK. esp8266 is EoL in support of espressif
